company-irony.el

```diff
@@ -73,7 +73,8 @@
           (irony-completion-candidates-async
            (lambda (candidates) ;; closure, lexically bound
              (funcall callback
-                      (company-irony--filter-candidates prefix candidates)))))))
+                      (company-irony--filter-candidates prefix candidates)))
```

Comment:
`company-irony-ignore-case` should be supported in the irony-server command, and the string-prefix-p done by `company-irony--filter-candidates` does not seem necessary any longer.
The name `filter` would no longer make sense, the new function would only propertize, so maybe something like `company-irony--make-candidates` would be better.